test/tracking.dart

```diff
@@ -0,0 +1,57 @@
+import 'package:mobx/mobx.dart';
+import 'package:mobx/src/api/context.dart';
+import 'package:mobx/src/core/tracking.dart';
+import 'package:test/test.dart';
+
+void main() {
+  test('Tracking reacts to changes to reactive values between begin and end',
```

Comment:
Can we also have a test with a autorun() to ensure the reactions are kicked off when the observables change outside